### DISCUSSION TITLE
Run cli and execute a default subcommand

### DISCUSSION THREAD

**Discussion by shaikhu** [remkop/picocli#2141] [Q&A] [answered]
Hello 👋 
For the simple example below, I have a cli tool with 2 subcommands.  This runs as expected e.g.`cli basic foo` or `cli advanced foo`.  

In hindsight (and with a proper design) the basic search subcommand should have been the main command. 
1. It is possible to run `Cli` with a default subcommand?  e.g. I would like `cli foo` to be treated as `cli basic foo`.  
2. I'd like `cli basic foo` to continue to work (as well as other subcommands in the future).

```
@CommandLine.Command(name = "cli", subcommands = {Cli.BasicSearch, Cli.AdvancedSearch}")
public class Cli {
  @CommandLine.Command(name = "basic", description = "runs a basic search")
  public class BasicSearch extends Callable<Integer> {.  
    @Override 
    public Integer call() {
      System.out.println("basic search");
      return 0;
    }
  }
 
 @CommandLine.Command(name = "advanced", description = "runs an advanced search")
  public class AdvancedSearch extends Callable<Integer> {
  @Override 
    public Integer call() {
      System.out.println("advanced search");
      return 0;
    } 
  }
}
```

Is there a better approach to what I'm trying to achieve?

**Comment by remkop** [comment, marked as answer]
One way to accomplish this is to let the top-level `cli` command delegate to the `basic` subcommand:

- make the top-level `cli` command implement `Callable<Integer>` just like the `basic` subcommand;
- copy all options and positional parameters from the `basic` subcommand to the top-level `cli` command
- optionally make these options and positional parameters [hidden](https://picocli.info/#_hidden_options_and_parameters) in the top-level `cli` command, so they don't show up in the usage help for the top-level command
- in the `call` method of the top-level command, delegate to the `basic` command (copy all options/params to an instance of `basic`, and call its `call` method)

**Reply by shaikhu** [reply]
Works great. Thank you!
